### ISSUE TITLE
[critical] worldmap panel not being registered as a plugin on Hosted Grafana instances

### ISSUE THREAD

**Issue by mattttt** [raintank/worldping-app#167]
I spun up a new HG with wP installed and the dashboards with world map are throwing an error. So plugin.json doesnt explicitly include the grafana-worldmap-plugin, however it is pulled in via grunt and tossed in  `/dist/grafana-worldmap-panel`, which loads just find when i build it locally - but does not appear to load in the HG version.

Tried to replicate locally using the following steps:

1. rm -rf /data/plugins/worldping-app
2. git clone https://github.com/raintank/worldping-app.git
3. Restarted Grafana (./bin/grafana-server)

Everything seems to register fine, including worldmap:
<img width="829" alt="screenshot 2016-12-30 19 28 53" src="https://cloud.githubusercontent.com/assets/2886187/21574731/3c237daa-cec6-11e6-8d0f-46dfc1ebc8d8.png">

@DanCech said on slack:
> "Hg doesn't do anything special,  it just runs grafana-cli plugins install X for each plugin"

So perhaps it's something to do with how grafana-cli runs vs these local dev instances?

Im also curious how @daniellee tested the app for this latest version, and whether he installed worldping via CLI tool or git clone.

**Comment by mattttt** [comment]
Note on the above - I've confirmed worldmap panel does not exist _anywhere_ else on my system. **Only** within the worldping-app directory.

**Comment by daniellee** [comment]
@mattttt I installed it via the CLI locally, not via git clone. I wanted to install the same way a customer would.

**Comment by mattttt** [comment]
@daniellee Cool, was hoping you'd say that. Which makes this even more confusing. Can you and @DanCech put your heads together and sort out what's going on here? 

I dont want to pre-install wP if people are going to be seeing errors.

**Comment by DanCech** [comment]
From test2.grafana.net:
```
t=2017-01-02T16:46:12+0000 lvl=info msg="Starting plugin search" logger=plugins
t=2017-01-02T16:46:12+0000 lvl=info msg="Registering plugin" logger=plugins name="Worldmap Panel"
t=2017-01-02T16:46:12+0000 lvl=info msg="Registering plugin" logger=plugins name="worldPing CTA"
t=2017-01-02T16:46:12+0000 lvl=info msg="Registering plugin" logger=plugins name="worldPing Endpoint List"
t=2017-01-02T16:46:12+0000 lvl=info msg="Registering plugin" logger=plugins name="worldPing Endpoint Nav"
t=2017-01-02T16:46:12+0000 lvl=info msg="Registering plugin" logger=plugins name=worldPing
```

**Comment by mattttt** [comment]
ok, does that mean it works? Was my experience on Friday night an anomaly?

**Comment by DanCech** [comment]
No, just gathering data.  The server discovers the plugin, but it doesn't work in the frontend.

I managed to snag this error message:
```
Error loading https://test2.grafana.net/public/plugins/raintank-worldping-app/grafana-worldmap-panel/libs/leaflet.js as "./libs/leaflet" from https://test2.grafana.net/public/plugins/raintank-worldping-app/grafana-worldmap-panel/worldmap.js
```

**Comment by DanCech** [comment]
From the server log:
```
2017/01/02 17:17:08 http: proxy error: net/http: request canceled
t=2017-01-02T17:17:08+0000 lvl=info msg="Request Completed" logger=context userId=3 orgId=1 uname=dcech method=GET path=/api/profile/org status=502 remote_addr=45.47.117.60 time_ms=96ns size=0
2017/01/02 17:17:22 http: multiple response.WriteHeader calls
t=2017-01-02T17:17:22+0000 lvl=eror msg="Request error" logger=context userId=3 orgId=1 uname=dcech error="session(release): Error 1213: Deadlock found when trying to get lock; try restarting transaction" stack="/usr/local/go/src/runtime/panic.go:458 (0x432bb3)\n/go/src/github.com/grafana/grafana/pkg/middleware/session.go:74 (0x624d9b)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:104 (0x5895a6)\n/go/src/github.com/grafana/grafana/pkg/middleware/recovery.go:146 (0x624a01)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:104 (0x5895a6)\n/go/src/github.com/grafana/grafana/pkg/middleware/logger.go:33 (0x622ac5)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/router.go:184 (0x599d2c)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/router.go:286 (0x5941ae)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/macaron.go:175 (0x58d84b)\n/usr/local/go/src/net/http/server.go:2202 (0x55f07d)\n/usr/local/go/src/net/http/server.go:1579 (0x55bbd7)\n/usr/local/go/src/runtime/asm_amd64.s:2086 (0x463171)\n"
t=2017-01-02T17:17:22+0000 lvl=eror msg="Request Completed" logger=context userId=3 orgId=1 uname=dcech method=POST path=/graphite/render status=500 remote_addr=45.47.117.60 time_ms=688ns size=25026
t=2017-01-02T17:17:22+0000 lvl=eror msg="Request error" logger=context userId=3 orgId=1 uname=dcech error="session(release): Error 1213: WSREP detected deadlock/conflict and aborted the transaction. Try restarting the transaction" stack="/usr/local/go/src/runtime/panic.go:458 (0x432bb3)\n/go/src/github.com/grafana/grafana/pkg/middleware/session.go:74 (0x624d9b)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:104 (0x5895a6)\n/go/src/github.com/grafana/grafana/pkg/middleware/recovery.go:146 (0x624a01)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:104 (0x5895a6)\n/go/src/github.com/grafana/grafana/pkg/middleware/logger.go:33 (0x622ac5)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/router.go:184 (0x599d2c)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/router.go:286 (0x5941ae)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/macaron.go:175 (0x58d84b)\n/usr/local/go/src/net/http/server.go:2202 (0x55f07d)\n/usr/local/go/src/net/http/server.go:1579 (0x55bbd7)\n/usr/local/go/src/runtime/asm_amd64.s:2086 (0x463171)\n"
t=2017-01-02T17:17:22+0000 lvl=eror msg="Request Completed" logger=context userId=3 orgId=1 uname=dcech method=POST path=/graphite/render status=500 remote_addr=45.47.117.60 time_ms=996ns size=791
2017/01/02 17:17:22 http: multiple response.WriteHeader calls
t=2017-01-02T17:17:33+0000 lvl=eror msg="Request error" logger=context userId=3 orgId=1 uname=dcech error="session(release): Error 1213: Deadlock found when trying to get lock; try restarting transaction" stack="/usr/local/go/src/runtime/panic.go:458 (0x432bb3)\n/go/src/github.com/grafana/grafana/pkg/middleware/session.go:74 (0x624d9b)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:104 (0x5895a6)\n/go/src/github.com/grafana/grafana/pkg/middleware/recovery.go:146 (0x624a01)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:104 (0x5895a6)\n/go/src/github.com/grafana/grafana/pkg/middleware/logger.go:33 (0x622ac5)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/router.go:184 (0x599d2c)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/router.go:286 (0x5941ae)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/macaron.go:175 (0x58d84b)\n/usr/local/go/src/net/http/server.go:2202 (0x55f07d)\n/usr/local/go/src/net/http/server.go:1579 (0x55bbd7)\n/usr/local/go/src/runtime/asm_amd64.s:2086 (0x463171)\n"
t=2017-01-02T17:17:33+0000 lvl=eror msg="Request Completed" logger=context userId=3 orgId=1 uname=dcech method=POST path=/graphite/render status=500 remote_addr=45.47.117.60 time_ms=862ns size=1061
t=2017-01-02T17:17:33+0000 lvl=eror msg="Request error" logger=context userId=3 orgId=1 uname=dcech error="session(release): Error 1213: WSREP detected deadlock/conflict and aborted the transaction. Try restarting the transaction" stack="/usr/local/go/src/runtime/panic.go:458 (0x432bb3)\n/go/src/github.com/grafana/grafana/pkg/middleware/session.go:74 (0x624d9b)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:104 (0x5895a6)\n/go/src/github.com/grafana/grafana/pkg/middleware/recovery.go:146 (0x624a01)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:104 (0x5895a6)\n/go/src/github.com/grafana/grafana/pkg/middleware/logger.go:33 (0x622ac5)\n/usr/local/go/src/runtime/asm_amd64.s:479 (0x46062c)\n/usr/local/go/src/reflect/value.go:434 (0x639d88)\n/usr/local/go/src/reflect/value.go:302 (0x6396a4)\n/go/src/github.com/grafana/grafana/vendor/github.com/go-macaron/inject/inject.go:117 (0x866975)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/context.go:113 (0x589682)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/router.go:184 (0x599d2c)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/router.go:286 (0x5941ae)\n/go/src/github.com/grafana/grafana/vendor/gopkg.in/macaron.v1/macaron.go:175 (0x58d84b)\n/usr/local/go/src/net/http/server.go:2202 (0x55f07d)\n/usr/local/go/src/net/http/server.go:1579 (0x55bbd7)\n/usr/local/go/src/runtime/asm_amd64.s:2086 (0x463171)\n"
t=2017-01-02T17:17:33+0000 lvl=eror msg="Request Completed" logger=context userId=3 orgId=1 uname=dcech method=POST path=/graphite/render status=500 remote_addr=45.47.117.60 time_ms=871ns size=1898
2017/01/02 17:17:33 http: multiple response.WriteHeader calls
2017/01/02 17:17:33 http: multiple response.WriteHeader calls
t=2017-01-02T17:17:43+0000 lvl=info msg="Request Completed" logger=context userId=3 orgId=1 uname=dcech method=GET path=/public/plugins/raintank-worldping-app/grafana-worldmap-panel/libs/leaflet.js status=404 remote_addr=45.47.117.60 time_ms=18ns size=2867
t=2017-01-02T17:17:57+0000 lvl=info msg="Request Completed" logger=context userId=3 orgId=1 uname=dcech method=GET path=/public/plugins/raintank-worldping-app/grafana-worldmap-panel/libs/leaflet.js status=404 remote_addr=45.47.117.60 time_ms=112ns size=2867
```

**Comment by mattttt** [comment]
WorldMap has been resolved by @DanCech, but he uncovered another issue where ES 1.x support has been removed from Grafana v4:
https://github.com/grafana/grafana/commit/99c5f7d571dd69169f09daf466e7a53046d2bee1

To resolve, @awoods says: "someone will need to rewrite https://github.com/raintank/tsdb-gw/blob/master/elasticsearch/elasticsearch.go to support the 2.x query payload"

Will open another issue to address this: https://github.com/raintank/worldping-app/issues/168

**Comment by mattttt** [comment]
Fixed in wp 1.2.1. Tested and released.
